docs/proposals/0000-support-interchangeable-build-and-run-implementations.md

```diff
@@ -0,0 +1,248 @@
+kind: BuildWorkload
+metadata:
+  name: my-build-guid
+  namespace: my-space-guid
+spec:
+  env:
+    - name: VCAP_SERVICES
+      valueFrom:
+        secretKeyRef:
+          key: VCAP_SERVICES
+          name: vcap-services-secret
+  buildpacks: [] # Use empty to indicate "default"?
+  stack: io.buildpacks.stacks.bionic
+  source:
+    registry:
+      image: localregistry-docker-registry.default.svc.cluster.local:30050/kpack/packages/eb60d1d0-5de1-4051-ac05-a4047f2b191a@sha256:987d6b085314a16ba5566fd0821d5a2f59e1a54ea11b8c0e2cd09309482d0481
+      imagePullSecrets:
+        - name: image-registry-credentials
+status:
+  conditions:
+  - status: 'False'
+    type: Staging
+  - status: 'True'
+    type: Succeeded
+  droplet:
+    ports: []
+    processTypes:
+      - command: node /workspace/server.js
+        type: web
+    registry:
+      image: localregistry-docker-registry.default.svc.cluster.local:30050/korifi-controllers/kpack/images/dbcee519-d239-4272-a264-3c1ce36a3e0a@sha256:6cee35c7d3ddc688bc45745e7c54432063221f71c6a092a649dcdff17a028c90
+      imagePullSecrets:
+        - name: image-registry-credentials
+    stack: io.buildpacks.stacks.bionic
+```
+
+```yaml
+---
+apiVersion: TBD.cloudfoundry.org/v1alpha1
+kind: RunWorkload
+metadata:
+  name: my-proc-guid
+  namespace: my-space-guid
+spec:
+  command:
+    - /cnb/lifecycle/launcher
+    - node /workspace/server.js
+  
+  resources:
+    limits:
+      ephemeral-storage: 1024M
+      memory: 1024M
+    requests:
+      cpu: '0'
+      memory: 1024M
+      
+  envFrom:
+    - secretRef:
+        name: user-specified-env-vars
+        
+  env:
+    - name: VCAP_APP_HOST
+      value: 0.0.0.0
+    - name: VCAP_APP_PORT
+      value: '8080'
+    - name: VCAP_SERVICES
+      valueFrom:
+        secretKeyRef:
+          key: VCAP_SERVICES
+          name: vcap-services-secret
+    - name: PORT
+      value: '8080'
+    - name: POD_NAME
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.name
+    - name: CF_INSTANCE_GUID
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.uid
+    - name: CF_INSTANCE_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.hostIP
+    - name: CF_INSTANCE_INTERNAL_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.podIP
+          
+  livenessProbe: #...
+  readinessProbe: #...
+  image: localregistry-docker-registry.default.svc.cluster.local:30050/cf-k8s-controllers/kpack/images/ff5b920f-c729-4284-9ae9-241d95b98528@sha256:7a7ed9b42905c6fabd71ea09eb22b97e0c638c9a341a70c84f9461e167c9f50f
+  imagePullSecrets:
+    - name: image-registry-credentials 
+  
+  instances: 1
+  
+  version: '0'
+```
+
+
+
+# Migration
+[migration]: #migration
+
+This will be a breaking change, but we haven't made any guarantee about upgrades. Existing installations will need to be
```

Comment:
I think we also want to rename the CRDs (take off the CF prefix) so it would be nice to coordinate this type of change with that.